### PR TITLE
#35265 - Tabs

### PR DESCRIPTION
**Pull request by kruchy8**
REDMINE: https://dev.snowdog.pro/issues/35265
ISSUE: https://github.com/SnowdogApps/magento2-alpaca-components/issues/91

### CODE REVIEW

File: components/02-elements/tab/_tab.scss [hunk 1]
```diff
@@ -1,99 +1,130 @@
...
+$tab__title-color-mobile-active         : $color-primary !default;
+$tab__title-font-weight                 : $font-weight-normal !default;
+$tab__title-font-weight--mobile         : $font-weight-bold !default;
+$tab__title-font-weight--active         : $font-weight-bold !default;
+$tab__title-background                  : $bg-color-base !default;
+$tab__title-background--active          : $gray-light !default;
+$tab__title-background--mobile-active   : $color-secondary !default;
+// content
+$tab__content-padding                   : $spacer--large $spacer--extra-large !default;
+$tab__content-padding--mobile           : $spacer--medium $spacer--large !default;
```
Comment (Igloczek) [inline components/02-elements/tab/_tab.scss]: It's mobile first, so you should add a modificator, to the larger screen and you shouldn't use `mobile` / `desktop` / `phone` / `tablet` and other words that describe device itself. Use `small` / `medium` / `large` instaed.

File: components/02-elements/tab/_tab.scss [hunk 2]
```diff
@@ -1,99 +1,130 @@
...
-                border-width: $tab__title-border-width--active;
-                border-style: $tab__title-border-style--active;
+                border-bottom: $tab__title-border--active;
                 background-color: $tab__title-background--active;
+                font-weight: $tab__title-font-weight--active;
+                color: $tab__title-color;
                 z-index: 2;
             }
+            .tab__icon {
+                transform: translateY(-50%)rotateX(180deg);
```
Comment (Igloczek) [inline components/02-elements/tab/_tab.scss]: Will be nice to have a space between transform rules

File: components/02-elements/tab/tab.js [hunk 3]
```diff
@@ -1,14 +1,17 @@
...
       activeContentClass = 'tab__content--active';
 
-
 tab.forEach(key => key.addEventListener('click', event => {
-  const self = event.target,
-        tabId = self.dataset.tab;
+  let self = event.target;
+
+  if (self.tagName === 'svg' || self.classList.contains('tab__icon')) {
+    self = self.closest('.tab__title');
```
Comment (Igloczek) [inline components/02-elements/tab/tab.js]: Use `event.currentTarget` instead just `target` and you will be able to remove this code.

Code sample - https://codepen.io/Igloczek/pen/ooxvGQ

File: components/02-elements/tab/_tab.scss [hunk 4]
```diff
@@ -1,99 +1,130 @@
...
+                }
             }
         }
     }
+
+    &__icon {
+        position: absolute;
+        right: 15px;
+        top: 50%;
+        transform: translateY(-50%);
```
Comment (Igloczek) [inline components/02-elements/tab/_tab.scss]: Let's use:
```
top: 0;
bottom: 0;
margin: auto;
```
it's a better way of vertical aligning of elements with known size.

File: components/02-elements/tab/_tab.scss (hunk 2 above)
Comment (kruchy8) [reply components/02-elements/tab/_tab.scss]: Done

File: components/02-elements/tab/_tab.scss (hunk 1 above)
Comment (kruchy8) [reply components/02-elements/tab/_tab.scss]: Done

File: components/02-elements/tab/_tab.scss (hunk 4 above)
Comment (kruchy8) [reply components/02-elements/tab/_tab.scss]: I would like to but...there is a slight problem :(
image
and if I disable `position: absolute` :
image
What do you think, Bartek ?

File: components/02-elements/tab/tab.js (hunk 3 above)
Comment (kruchy8) [reply components/02-elements/tab/tab.js]: great ! thanks ;)

File: components/02-elements/tab/_tab.scss (hunk 4 above)
Comment (Igloczek) [reply components/02-elements/tab/_tab.scss]: It works, you just have to remove the `translateY` from modificator and keep the `right` property.

File: components/02-elements/tab/tab.js [hunk 5]
```diff
@@ -1,14 +1,14 @@
...
 const tab                = document.querySelectorAll('.tab__title'),
       content            = document.querySelectorAll('.tab__content'),
       activeTitleClass   = 'tab__title--active',
       activeContentClass = 'tab__content--active';
 
-
 tab.forEach(key => key.addEventListener('click', event => {
-  const self = event.target,
-        tabId = self.dataset.tab;
+  let self = event.currentTarget;
```
Comment (Igloczek) [inline components/02-elements/tab/tab.js]: Now it can be `const` and you can combine all variables together.

File: CHANGELOG.md [hunk 6]
```diff
@@ -5,7 +5,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
-
+ - Tab styling that fits alpaca design.
```
Comment (Igloczek) [inline CHANGELOG.md]: Heading missing.

File: components/02-elements/tab/_tab.scss (hunk 4 above)
Comment (kruchy8) [reply components/02-elements/tab/_tab.scss]: you're right. Done

File: components/02-elements/tab/tab.js (hunk 5 above)
Comment (kruchy8) [reply components/02-elements/tab/tab.js]: Done

File: CHANGELOG.md (hunk 6 above)
Comment (kruchy8) [reply CHANGELOG.md]: Fixed.

**Pull request merged**